apps/surfforecast/surf_forecast.star

```diff
@@ -0,0 +1,514 @@
+    darkness_width = get_darkness_width(state)
+    return render.Row(
+        children = [
+            render.Box(
+                height = TIDBYT_HEIGHT,
+                width = first_light_width,
+                color = COLORS["DAYLIGHT"],
+            ),
+            render.Box(
+                height = TIDBYT_HEIGHT,
+                width = DUSK_WIDTH,
+                color = COLORS["DUSK"],
+            ),
+            render.Box(
+                height = TIDBYT_HEIGHT,
+                width = darkness_width,
+                color = COLORS["BLACK"],
+            ),
+            render.Box(
+                height = TIDBYT_HEIGHT,
+                width = DUSK_WIDTH,
+                color = COLORS["DUSK"],
+            ),
+            render.Box(
+                height = TIDBYT_HEIGHT,
+                width = second_light_width,
+                color = COLORS["DAYLIGHT"],
+            ),
+        ],
+    )
+
+def render_bars(state):
+    return render.Padding(
+        pad = (1, 0, 1, 0),
+        child = render.Column(
+            expanded = True,
+            main_align = "end",
+            children = [
+                render.Row(
+                    cross_align = "end",
+                    main_align = "space_between",
+                    expanded = True,
+                    children = [render_bar(d) for d in get_bar_data(state)],
+                ),
+            ],
+        ),
+    )
+
+def render_bar(data):
+    if data["IS_CURRENT"]:
+        return render_flashing_bar(data)
+
+    return render.Column(
+        cross_align = "center",
+        children = [
+            render.Box(width = 4, height = data["MAX_BAR_HEIGHT"], color = COLORS["MAX_BAR"]),
+            render.Box(width = 4, height = data["MIN_BAR_HEIGHT"], color = COLORS["MIN_BAR"]),
+            render.Box(width = 6, height = 2, color = data["RATING_COLOR"]),
+        ],
+    )
+
+def render_flashing_bar(data):
+    return render.Animation(
+        children = [
+            render.Column(
+                cross_align = "center",
+                children = [
+                    render.Box(width = 4, height = data["MAX_BAR_HEIGHT"], color = transparent(COLORS["MAX_BAR"], p)),
+                    render.Box(width = 4, height = data["MIN_BAR_HEIGHT"], color = transparent(COLORS["MIN_BAR"], p)),
+                    render.Box(width = 6, height = 2, color = transparent(data["RATING_COLOR"], p)),
+                ],
+            )
+            for p in get_animation_percentages()
+        ],
+    )
+
+def get(url):
+    rep = http.get(url)
+    if rep.status_code != 200:
+        fail("Surfline request failed with status %d", rep.status_code)
+    return rep.json()
+
+def get_response(url, cache_key):
+    response_cached = cache.get(cache_key)
+    if response_cached != None:
+        return json.decode(response_cached)
+
+    response = get(url)
+    cache.set(cache_key, json.encode(response), ttl_seconds = 240)
```

Comment:
Updated! Good catch. Since this app displays live conditions as well as a forecast, I set the TTL to 15 minutes for requests related to live conditions and 1 hour for requests related to the forecasts.